### PR TITLE
Sprint3 search fixes

### PR DESCRIPTION
**Pull request by TiruDeepthi**

### CODE REVIEW

File: src/components/__test__/GlobalSearch.test.js [hunk 1]
```diff
@@ -1,11 +1,75 @@
...
+    const input = screen.getByPlaceholderText(
+      "Search by Style or Title or Brand"
+    );
+    return {
+      input,
+      ...utils
+    };
+  };
+
+  it("OnChange", () => {
```
Comment (RajKumar-BackCountry) [inline src/components/__test__/GlobalSearch.test.js]: Make it meaningful.
On Changing value of text areas, the value should be reflected as expected.

Do the same for other test-cases also.

File: src/components/__test__/GlobalSearch.test.js [hunk 2]
```diff
@@ -1,11 +1,75 @@
...
+    expect(input.value).toBe("test");
+    fireEvent.change(input, { target: { value: "" } });
+    expect(input.value).toBe("");
+  });
+
+  it("onClick", () => {
+    const onClick = jest.fn();
+    render(<GlobalSearch onClick={onClick}/>);
+    fireEvent.click(screen.getByRole("button", /Search/i));
+    expect(onClick).toHaveBeenCalledTimes(0);
```
Comment (RajKumar-BackCountry) [inline src/components/__test__/GlobalSearch.test.js:82]: What are u achieving by this test case ?

File: src/components/__test__/GlobalSearch.test.js [hunk 3]
```diff
@@ -1,11 +1,75 @@
...
+  });
+});
```
Comment (RajKumar-BackCountry) [inline src/components/__test__/GlobalSearch.test.js:108]: Where are the other test cases ?

1. On clicking cross-icon, input box should be cleared.
2. On clicking submit, the loader should be appear and the corresponding api should be called.
3. Loader should disappear and search results should be shown.

File: src/components/__test__/GlobalSearch.test.js [hunk 4]
```diff
@@ -1,11 +1,108 @@
...
+    render(<GlobalSearch onClick={onClick} />);
+    fireEvent.click(screen.getByRole("button", /Search/i));
+    expect(onClick).toHaveBeenCalledTimes(0);
+  });
+
+  it("API call on Search click", async() => {
+    const { input } = setup();
+    fireEvent.change(input, { target: { value: "test" } })
+    fireEvent.click(screen.getByText("Search"))
+    jest.spyOn(global, "fetch").mockResolvedValue({
```
Comment (RajKumar-BackCountry) [inline src/components/__test__/GlobalSearch.test.js:89]: where are you checking if the mock api is being called ?

Comment (TiruDeepthi) [reply src/components/__test__/GlobalSearch.test.js:89]: I am not getting way to call this mock api

**Comment by RajKumar-BackCountry** [comment]
Deepti is facing difficulities with handling the mock apis during test cases.
In the interest of time, I am approving the MR. But the test-cases are not completely done.

**Pull request merged**